### DISCUSSION TITLE
Many questions about using NICEGUI

### DISCUSSION THREAD

**Discussion by zouzhe1** [zauberzeug/nicegui#3349] [Q&A] [answered]
### Question

Question 1: There are only some examples, without a systematic summary of the different APIs.

Question 2: Since nicegui is for Python, I am developing a ROS2 robot using Python, and I have no understanding of front-end at all. Your website documentation seems to assume that I already know front-end. The greatest value of your nicegui is to allow people who don't know front-end to create web interfaces! So how should this documentation be approached?

Question 3: When I import a 3D STL, the loading time is particularly slow. My file is only a 2KB cuboid, and my computer has an RTX 4090 + i7 13700k.

Question 4: I used app.add_static_files('/static', '/xx/xx/xx/resource') for the STL position, and the absolute path works, but within ROS2, I want to use a relative path, and I'm confused.

**Comment by python-and-novella** [comment, marked as answer]
**Answer**
A1: Reference of each control you could see it,which locates at the bottom of page.Maybe you need more time to learn documentation?
A2:Indeed,some styles need you know css\quasar\tailwind. But, nicegui is not a totally Python Framework, it's a quaser based framework by python. Some features of quasar\html that need you learn.It's inevitable.
If you want to ceate simple gui,you don't need to know css(not front-end,base layout belongs to front-end).
If you want to ceate beautiful persional gui, you should learm lots of front-end.
A3A4:Need code to reproduce,create a new discussion plz.
